itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -399,6 +399,42 @@ public void testIncrementalMetadataReplication() throws Throwable {
   }
 
   @Test
+  public void testNonReplDBMetadataReplication() throws Throwable {
+    String dbName = primaryDbName + "_metadata";
+    WarehouseInstance.Tuple tuple = primary
+            .run("create database " + dbName)
+            .run("use " + dbName)
+            .run("create table table1 (i int, j int)")
+            .run("create table table2 (a int, city string) partitioned by (country string)")
+            .run("create table table3 (i int, j int)")
+            .run("insert into table1 values (1,2)")
+            .dump(dbName, null, Arrays.asList("'hive.repl.dump.metadata.only'='true'"));
```

Comment:
DAS uses warehouse level dump. Shall check if warehouse metadata-only dump works without repl.source.for.